query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -72,7 +73,7 @@ private Duration setTimeSeriesPeriod(QueryRequest request) {
         }
       }
     }
-    return period;
+    return period.isZero() ? Optional.empty() : Optional.of(period);
```

Comment:
minor nit - even this is unnecessary, but less problematic because it's contained in the function. Could return `Optional.of(parseDuration(timeSeriesPeriod));` on line 72 and then on line 76, return empty optional if reached. It of course has its place and is sometimes required, but _any time_ you reassign a variable, you should consider alternatives.